packages/headless/src/controllers/triggers/headless-query-trigger.test.ts

```diff
@@ -0,0 +1,91 @@
+import {QueryTrigger, buildQueryTrigger} from './headless-query-trigger';
+import {
+  buildMockSearchAppEngine,
+  MockSearchEngine,
+} from '../../test/mock-engine';
+import {triggers, configuration} from '../../app/reducers';
+import {updateQuery} from '../../features/query/query-actions';
+import {executeSearch} from '../../features/search/search-actions';
+import {clearQueryTrigger} from '../../features/triggers/trigger-actions';
+
+describe('QueryTrigger', () => {
+  let engine: MockSearchEngine;
+  let queryTrigger: QueryTrigger;
+
+  function initQueryTrigger() {
+    queryTrigger = buildQueryTrigger(engine);
+  }
+
+  function getUpdateQueryAction() {
+    return engine.actions.find((a) => a.type === updateQuery.type);
+  }
+
+  function getClearQueryTriggerAction() {
+    return engine.actions.find((a) => a.type === clearQueryTrigger.type);
+  }
+
+  function registeredListeners() {
+    return (engine.subscribe as jest.Mock).mock.calls.map((args) => args[0]);
+  }
+
+  beforeEach(() => {
+    engine = buildMockSearchAppEngine();
+    initQueryTrigger();
+  });
+
+  it('initializes', () => {
+    expect(queryTrigger).toBeTruthy();
+  });
+
+  it('it adds the correct reducers to the engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      triggers,
+      configuration,
+    });
+  });
+
+  it('exposes a #subscribe method', () => {
+    expect(queryTrigger.subscribe).toBeTruthy();
+  });
+
+  it('when the #engine.state.triggers.query is not updated, it does not dispatch #clearQueryTrigger, #updateQuery, #executeSearch, and #logQueryTrigger', () => {
```

Comment:
If behaviour is different before and after we subscribe to the controller's state, I think it 
may be a good idea to add a "before subscribing" test.